base/shell/explorer/startup.cpp

```diff
@@ -85,8 +85,37 @@ static int runCmd(LPWSTR cmdline, LPCWSTR dir, BOOL wait, BOOL minimized)
     TRACE("Successfully ran command\n");
 
     if (wait)
-    {   /* wait for the process to exit */
-        WaitForSingleObject(info.hProcess, INFINITE);
+    {
+        HANDLE Handles[] = { info.hProcess };
+        DWORD nCount = _countof(Handles);
+        DWORD dwWait;
+        MSG msg;
+
+        /* wait for the process to exit */
+        for (;;)
+        {
+            /* We need to keep processing messages,
+               otherwise we will hang anything that is trying to send a message to us */
+            dwWait = MsgWaitForMultipleObjects(nCount, Handles, FALSE, INFINITE, QS_ALLINPUT);
```

Comment:
Why not just using
```c
MsgWaitForMultipleObjects(1, &info.hProcess, FALSE, INFINITE, QS_ALLINPUT);
```
since you're waiting on a single handle?

Reply:
I like this better.